File: c++/functions/function_passing_to_function.cpp
/*
 * Copyright (C) 2019-2026 Katarzyna Krasińska
 * Languagium - https://github.com/katheroine/languagium
 * Licensed under GPL-3.0 - see LICENSE.md
 */
// SPDX-License-Identifier: GPL-3.0-or-later

#include <iostream>

const char* inside()
{
  std::cout << "* Inside.\n";
  return "IN";
}

const char* outside(const char* (*function)())
{
  std::cout << "# Outside:\n"
    "# Calling passed function...\n";
  const char* result = function();
  std::cout << "# result: " << result << std::endl;
  return "OUT";
}

int main()
{
  std::cout << "Calling function...\n";
  const char* result = outside(inside);
  std::cout << "result: " << result << std::endl;
}

File: javascript/functions/function_passing_to_function.js
/*
 * Copyright (C) 2019-2026 Katarzyna Krasińska
 * Languagium - https://github.com/katheroine/languagium
 * Licensed under GPL-3.0 - see LICENSE.md
 */

function inside()
{
  console.log("* Inside.");
  return "IN";
}

function outside(some_function)
{
  console.log("# Outside:");
  console.log("# Calling passed function...");
  let result = some_function();
  console.log("# result: " + result);
  return "OUT";
}

console.log("Calling function...");
let result = outside(inside);
console.log("result: " + result);
